datetime format in administration

PheysX opened the pull request:
<!--
Thank you for contributing to Shopware! Please fill out this description template to help us to process your pull request.

Please make sure to fulfil our contribution guideline (https://docs.shopware.com/en/shopware-platform-dev-en/contribution/contribution-guideline?category=shopware-platform-dev-en/contribution).

Do your changes need to be mentioned in the documentation?
Add notes on your change right now in the documentation files in /src/Docs/Resources and add them to the pull request as well. 
-->

### 1. Why is this change necessary?
* The orderDateTime in `sw-customer-detail-order.html.twig` is not the orderDateTime and its not formatted.
* Show hours and minutes in datetimes to give the shopowner in this views more control and a more specific result.



### 2. What does this change do, exactly?
* Changed the `createdAt` column to `orderDateTime` in `src/Administration/Resources/app/administration/src/module/sw-customer/view/sw-customer-detail-order/sw-customer-detail-order.html.twig`
* Formatted dateTimes to show hours and minutes in:
  * `src/Administration/Resources/app/administration/src/module/sw-customer/component/sw-customer-base-info/sw-customer-base-info.html.twig`
  * `src/Administration/Resources/app/administration/src/module/sw-customer/view/sw-customer-detail-order/sw-customer-detail-order.html.twig`
  * `src/Administration/Resources/app/administration/src/module/sw-order/component/sw-order-user-card/index.js`

### 3. Describe each step to reproduce the issue or behaviour.


### 4. Please link to the relevant issues (if any).


### 5. Checklist

- [ ] I have written tests and verified that they fail without my change
- [ ] I have squashed any insignificant commits
- [x] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/2020-08-03-Implement-New-Changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

PheysX committed:
Updated datetimes

PheysX committed:
Added changelog

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/shopware/platform?pullRequest=2104) <br/>All committers have signed the CLA.

jleifeld commented:
Can you try to just use the date filter? Normally it should also show the time

PheysX committed:
Adjust birthday, removed optional date parameters

PheysX commented:
> Can you try to just use the date filter? Normally it should also show the time

You are right. Changed the keywords in changelog and removed birthday hours and minutes aswell.

Without my changes:
![image](https://user-images.githubusercontent.com/28557712/136509785-31bcca65-f358-4c13-bc59-3d46cf45c3d2.png)

mitelg commented on src/Administration/Resources/app/administration/src/module/sw-customer/view/sw-customer-detail-order/sw-customer-detail-order.html.twig:
```diff
@@ -26,25 +26,25 @@
+                </template>
+                {% endblock %}
 
-            {% block sw_customer_detail_order_card_grid_columns_date %}
```
do not remove blocks.
you could leave them empty and mark them as deprecated

PheysX committed:
Mark old block as deprecated

Haberkamp commented on src/Administration/Resources/app/administration/src/module/sw-customer/component/sw-customer-base-info/sw-customer-base-info.html.twig:
```diff
@@ -220,7 +225,7 @@
             <dd v-if="!customerEditMode">
                 <template v-if="customer.birthday">
-                    {{ customer.birthday | date }}
+                    {{ customer.birthday | date({ minute: undefined, hour: undefined }) }}
```
Please just use the date filter without any parameters. Because you cannot specify a date-time through the administration.

PheysX replied:
Without parameters, we will see hours and minutes on a birth**day** field. Instead of 01.01.2021 we would have in germany in the moment: 01.01.2021, 02:00 like i mentioned with the picture earlier in this thread.

It that on purpose?

Haberkamp replied:
Ahh, yes my bad. Browser caching problem. 😅 . It would be nice if you could add jest tests for your changes. The rest looks fine to me.

PheysX replied:
For what exactly? For the output of date filter function with this parameters? Never done this before.

Haberkamp replied:
The date filter is already tested, but you could add a test that renders the component and that makes sure the date is displayed. To prevent others from accidentally break the implementation on that page.

PheysX commented on src/Administration/Resources/app/administration/src/module/sw-customer/view/sw-customer-detail-order/sw-customer-detail-order.html.twig:
```diff
@@ -26,25 +26,28 @@
+
+                {% block sw_customer_detail_order_card_grid_columns_order_date_time %}
+                <template #column-orderDateTime="{ item }">
+                    {{ item.orderDateTime }}
```
Oh, here is a mistake. No date filter anymore. But, when i have to change this, should i use the new `<sw-time-ago :date="item.orderDateTime" />`  component like in sw-order-list instead? @Haberkamp

PheysX replied:
Added the shopware component `sw-time-ago`.

PheysX committed:
Added sw-time-ago component and  jest tests

PheysX commented:
Hope that jest test is fine :D

PheysX commented:
Hey @King-of-Babylon, do you have a new status for me?

King-of-Babylon commented:
Hi @PheysX,
yes :tada: Your change is now merged: https://github.com/shopware/platform/commit/2b0d995bd6e6f0bf31142e01bdfaa5aefa9c1b0a
Thank you for your distribution :blue_heart: 
Now you can add a jest test to: https://github.com/shopware/platform/pull/2107 :wink:
